### PR TITLE
[Translation] improved exception when missing required component

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #13194 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by aitboudad** [comment]
:+1:

File: src/Symfony/Component/Translation/Loader/IcuResFileLoader.php [hunk 1]
```diff
@@ -36,6 +36,10 @@ public function load($resource, $locale, $domain = 'messages')
             throw new NotFoundResourceException(sprintf('File "%s" not found.', $resource));
         }
 
+        if (!class_exists('Symfony\Component\Config\Resource\DirectoryResource')) {
+            throw new \LogicException('Using the IcuResFileLoader requires the Symfony Config component.');
+        }
+
         try {
             $rb = new \ResourceBundle($locale, $resource);
```
Comment (aitboudad) [inline src/Symfony/Component/Translation/Loader/IcuResFileLoader.php]: can we check here if ```ResourceBundle``` exist as it's require ```symfony/intl```

Comment (xabbuh) [inline src/Symfony/Component/Translation/Loader/IcuResFileLoader.php]: Is that really necessary?

Comment (aitboudad) [inline src/Symfony/Component/Translation/Loader/IcuResFileLoader.php]: not really, leave as it is

**Comment by aitboudad** [comment]
the use of `FileResource` in catalogue is optional, so instead of throwing an exception what do you think to avoid `addResource` call ?

```php
if (class_exists('Symfony\Component\Config\Resource\FileResource')) {
    $catalogue->addResource(new FileResource($resource));
}
```

**Comment by xabbuh** [comment]
@aitboudad Is this what you had in mind?

**Comment by aitboudad** [comment]
@xabbuh yes thanks !

**Comment by aitboudad** [comment]
Thank you @xabbuh.

**Pull request merged**